cvmfs/cvmfs_server

```diff
@@ -4046,8 +4052,20 @@ snapshot() {
       -i ${spool_dir}/tmp/last_snapshot \
       -o .cvmfs_last_snapshot"
 
-    # remove magic file after successful replication
-    $user_shell "$delete_command"
+    # run the automatic garbage collection (if configured)
+    if is_garbage_collectable $name && [ ! -z "$CVMFS_GC_TIMESTAMP_THRESHOLD" ]
```

Comment:
> For newly created repositories with enabled garbage collection (cvmfs_server mkfs -z) this config variable defaults to 3 days ago. 

That default is only used on repository servers, in mkfs() and import() functions, not on stratum 1s' add-replica().  There's no default for new replicas on stratum 1s. 

> I don't think that defaulting CVMFS_GC_TIMESTAMP_THRESHOLD here is a good idea as there could be repositories that are garbage collectable (CVMFS_GARBAGE_COLLECTION=true) but don't want to make use of the automatic garbage collection feature (CVMFS_GC_TIMESTAMP_THRESHOLD is set in the configuration).

Oh, so you're overloading the meaning of "threshold" to also mean don't do any collection if it isn't set, and requiring stratum 1 administrators to set a value in order to get collection done.  The CVMFS_GARBAGE_COLLECTION flag is only indicating that it is collectable, not that collection is happening.  That's a way to do it, but it seems misleading to me.  Why bother to have the CVMFS_GARBAGE_COLLECTION flag at all if it doesn't do anything but make it possible to use the threshold?  I'd rather have add-replica() not set CVMFS_GARBAGE_COLLECTION=true on collectible repos and have a default for the threshold.